ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -156,7 +156,7 @@ public Statement build(final SingleStatementContext statement) {
     private final DataSourceExtractor dataSourceExtractor;
     private final MetaStore metaStore;
 
-    private int selectItemIndex = 0;
+    private int selectItemIndex = 100 + (new java.util.Random()).nextInt(800);
```

Comment:
This won't actually fix the issue, only mask it. There is still a chance that there will be a clash. 

It also makes the naming of columns non-deterministic.

A better fix would be to programatically detect clashes and increment the count on a clash and only on a clash.

Reply:
My problem is: I don't know how to detect existing names from the parser. The name's are assigned by the parser, but the Exception is thrown by the engine.